src/integ_test/java/games/strategy/engine/lobby/server/db/UserControllerIntegrationTest.java

```diff
@@ -72,86 +55,80 @@ private static void testGet(final DbTestConnection dbTestConnection) throws Exce
 
   @Test
   public void testDoesUserExist() {
-    testDoesUserExist(POSTGRES);
-  }
-
-  private static void testDoesUserExist(final DbTestConnection dbTestConnection) {
-    final DBUser user = givenUser();
-
-    dbTestConnection.controller.createUser(user, new HashedPassword(MD5Crypt.crypt(user.getName())));
-
-    assertTrue(dbTestConnection.controller.doesUserExist(user.getName()));
-  }
-
-  @Test
-  public void testDoesUserExistPostgres() {
-    testDoesUserExist(POSTGRES);
+    assertTrue(controller.doesUserExist(createUserWithMd5CryptHash().getName()));
+    assertTrue(controller.doesUserExist(createUserWithBCryptHash().getName()));
   }
 
   @Test
   public void testCreateDupe() throws Exception {
-    testCreateDupe(POSTGRES);
-  }
-
-  private static void testCreateDupe(final DbTestConnection dbTestConnection) throws Exception {
-    final DBUser user = givenUser();
-
-    dbTestConnection.controller.createUser(user, new HashedPassword(MD5Crypt.crypt(user.getName())));
-
-    try {
-      dbTestConnection.controller.createUser(user, new HashedPassword(MD5Crypt.crypt(user.getName())));
-      fail("Should not be allowed to create a dupe user");
-    } catch (final Exception expected) {
-      // expected
-    }
+    catchException(() -> controller.createUser(createUserWithMd5CryptHash(),
+        new HashedPassword(MD5Crypt.crypt(Util.createUniqueTimeStamp()))));
+    assertThat("Should not be allowed to create a dupe user", caughtException(), is(not(equalTo(null))));
   }
 
   @Test
   public void testLogin() throws Exception {
-    testLogin(POSTGRES);
-  }
-
-  private static void testLogin(final DbTestConnection dbTestConnection) throws Exception {
-    final DBUser user = givenUser();
-    final HashedPassword password = new HashedPassword(MD5Crypt.crypt(user.getName()));
-    dbTestConnection.controller.createUser(user, password);
-
-    assertTrue(dbTestConnection.controller.login(user.getName(), password));
+    final String password = MD5Crypt.crypt(Util.createUniqueTimeStamp());
+    final DBUser user = createUserWithHash(password, s -> s);
+    controller.updateUser(user,
+        new HashedPassword(BCrypt.hashpw(RsaAuthenticator.hashPasswordWithSalt(password), BCrypt.gensalt())));
```

Comment:
You can use your new helpers to simplify this statement:

```java
controller.updateUser(user, new HashedPassword(bcrypt(obfuscate(password))));
```

Reply:
Originally meant to do that  ^^
Seems like I missed it here